sdk/storage/azure_storage_blob/src/clients/append_blob_client.rs

```diff
@@ -40,45 +94,81 @@ impl AppendBlobClient {
-    pub fn endpoint(&self) -> &Url {
-        &self.endpoint
+    /// Creates a new AppendBlobClient, without providing any authentication information.
+    ///
+    /// # Arguments
+    ///
+    /// * `endpoint` - The full URL of the Azure storage account, for example `https://myaccount.blob.core.windows.net/`
+    /// * `container_name` - The name of the container containing this Append blob.
+    /// * `blob_name` - The name of the Append blob to interact with.
+    /// * `options` - Optional configuration for the client.
+    pub fn with_no_credential(
+        endpoint: &str,
+        container_name: &str,
+        blob_name: &str,
+        options: Option<AppendBlobClientOptions>,
+    ) -> Result<Self> {
+        let mut url = Url::parse(endpoint)?;
+
+        url.path_segments_mut()
+            .expect("Invalid endpoint URL: Cannot append container_name and blob_name to the blob endpoint.")
+            .extend([container_name, blob_name]);
+
+        let client = GeneratedAppendBlobClient::from_url(url, None, options)?;
+        Ok(Self { client })
     }
 
-    /// Gets the container name of the Storage account this client is connected to.
-    pub fn container_name(&self) -> &str {
-        &self.client.container_name
+    /// Creates a new AppendBlobClient from a blob URL.
+    ///
+    /// # Arguments
+    ///
+    /// * `blob_url` - The full URL of the Append blob, for example `https://myaccount.blob.core.windows.net/mycontainer/myblob`.
+    /// * `credential` - An optional implementation of [`TokenCredential`] that can provide an Entra ID token to use when authenticating.
+    /// * `options` - Optional configuration for the client.
+    pub fn from_blob_url(
+        blob_url: Url,
+        credential: Arc<dyn TokenCredential>,
+        options: Option<AppendBlobClientOptions>,
+    ) -> Result<Self> {
+        let client = GeneratedAppendBlobClient::from_url(blob_url, Some(credential), options)?;
+
+        Ok(Self { client })
+    }
+
+    /// Creates a new AppendBlobClient from a blob URL containing the SAS (shared access signature) query parameters.
+    ///
+    /// # Arguments
+    ///
+    /// * `blob_url` - The full URL of the Append blob, including the SAS query parameters.
+    /// * `credential` - An optional implementation of [`TokenCredential`] that can provide an Entra ID token to use when authenticating.
+    /// * `options` - Optional configuration for the client.
+    pub fn from_blob_sas_url(
+        blob_url: Url,
+        options: Option<AppendBlobClientOptions>,
+    ) -> Result<Self> {
+        let client = GeneratedAppendBlobClient::from_url(blob_url, None, options)?;
+
+        Ok(Self { client })
     }
 
-    /// Gets the blob name of the Storage account this client is connected to.
-    pub fn blob_name(&self) -> &str {
-        &self.client.blob_name
+    /// Gets the URL of the Storage account this client is connected to.
```

Comment:
I know you didn't write this in this PR but this comment is a bit off. Maybe something like "Gets the URL of the resource this client is configured for" - applies to all clients.